### PR TITLE
Adding LegacyTrackerTopology

### PR DESCRIPTION
**Pull request by alesaggio**

### CODE REVIEW

File: CalibTracker/SiStripLorentzAngle/python/EnsembleCalibrationLA_cfg.py [hunk 1]
```diff
@@ -3,10 +3,13 @@
 process = cms.Process("MACRO")
 process.add_(cms.Service("MessageLogger"))
 process.source = cms.Source("EmptySource")
+
```
Comment (pieterdavid) [inline CalibTracker/SiStripLorentzAngle/python/EnsembleCalibrationLA_cfg.py]: I would remove this empty line again, it's nicer to keep all these together

File: CalibTracker/SiStripLorentzAngle/src/LA_Filler.cc [hunk 2]
```diff
@@ -76,7 +73,7 @@ poly<std::string> LA_Filler_Fitter::
 granularity(const SiStripDetId detid, const float tthetaL, const Long64_t TFE_index, const float localy, const unsigned apvstrip) const {
   poly<std::string> gran;
   gran += subdetLabel(detid);
-  if(byLayer_)  gran *= layerLabel(detid);
+  if(byLayer_)  gran *= layerLabel(detid);
```
Comment (pieterdavid) [inline CalibTracker/SiStripLorentzAngle/src/LA_Filler.cc]: another whitespace detail: there's a space at the end of this line

File: DataFormats/TrackerCommon/interface/LegacyTrackerTopology.h [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        pxf.bladeMask_ =  0x3F;
+        pxf.panelMask_ =  0x3;
+        pxf.moduleMask_ = 0x3F;
+
+        TrackerTopology::TECValues tecVals;
+        tecVals.sideStartBit_ =        18;
+        tecVals.wheelStartBit_ =       14;
+        tecVals.petal_fw_bwStartBit_ = 12;
+        tecVals.petalStartBit_ =       8;
+	tecVals.ringStartBit_ =        5;
```
Comment (pieterdavid) [inline DataFormats/TrackerCommon/interface/LegacyTrackerTopology.h]: there seems to be something weird with the indentation here... maybe a mix of tabs and spaces?

File: DataFormats/TrackerCommon/interface/LegacyTrackerTopology.h [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        tobVals.rodStartBit_ =       5;
+        tobVals.moduleStartBit_ =    2;
+        tobVals.sterStartBit_ =      0;
+        tobVals.layerMask_ =     0x7;
+        tobVals.rod_fw_bwMask_ = 0x3;
+        tobVals.rodMask_ =       0x7F;
+        tobVals.moduleMask_ =    0x7;
+        tobVals.sterMask_ =      0x3;
+
+        std::unique_ptr<TrackerTopology> tTopo(new TrackerTopology(pxb, pxf, tecVals, tibVals, tidVals, tobVals));
```
Comment (pieterdavid) [inline DataFormats/TrackerCommon/interface/LegacyTrackerTopology.h]: I would simply write this as
```
return std::unique_ptr<TrackerTopology>{new TrackerTopology(pxb, pxf, tecVals, tibVals, tidVals, tobVals)};
```
but it's fine like this as well (unless C++14 is already required in cmssw, in that case we should go for `std::make_unique` @OlivierBondu , do you have any idea where to find that? I searched hypernews but didn't find the answer)

File: DataFormats/TrackerCommon/interface/LegacyTrackerTopology.h [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+#define TRACKER_TOPOLOGY_LEGACY_H
+
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "DataFormats/DetId/interface/DetId.h"
+
+namespace LegacyTrackerTopology {
+  
+  inline static std::unique_ptr<TrackerTopology> getTrackerTopology()
+  {
+        TrackerTopology::PixelBarrelValues pxb;
```
Comment (pieterdavid) [inline DataFormats/TrackerCommon/interface/LegacyTrackerTopology.h]: Shall we, for consistency, change these names into `pxbVals` and `pxfVals`?

**Review by pieterdavid — COMMENTED** [review]
Looks good to me :-) , only a few whitespace and detail comments.
One thing: when updating your pull request for cmssw, could you split this into two commits? One to introduce the `LegacyTrackerTopology` header, and another one with everything else (I think it makes the git history cleaner).

**Review by pieterdavid — APPROVED** [review]

**Review by OlivierBondu — APPROVED** [review]
Looks good to me! Indeed adding the `LegacyTrackerTopology` makes things look cleaner in macros...

One potential worry: there is nothing preventing people using this instead of accessing the topology throught the event setup?

I wonder if the `Legacy` in the file name should be in itself enough of a warning?

**Comment by alesaggio** [comment]
Well... it may not be enough indeed. Should I put a warning in LegacyTrackerTopology.h? Saying that whenever TrackerTopology is accessible from an EventSetup that header should not be used.

**Comment by pieterdavid** [comment]
Good idea. We could even consider triggering a compiler warning whenever the header is included - but that's all easy to add, and I'm sure we will hear some opinions once that goes into the cmssw PR ;)

**Comment by OlivierBondu** [comment]
:+1: for the written text warning in the header, for the rest indeed I'm sure we'll receive some comments during the PR!

**Comment by alesaggio** [comment]
If you're fine with the comment we can squash and merge ;)

File: DataFormats/TrackerCommon/interface/LegacyTrackerTopology.h [hunk 6]
```diff
@@ -4,6 +4,12 @@
 #include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
 #include "DataFormats/DetId/interface/DetId.h"
 
+
+// WARNING: this header has been introduced to call a TrackerTopology
+// object whenever it is not possible to access it trough an EventSeup.
```
Comment (OlivierBondu) [inline DataFormats/TrackerCommon/interface/LegacyTrackerTopology.h]: trough -> through
EventSeup -> EventSetup 

Thanks!

Comment (pieterdavid) [reply DataFormats/TrackerCommon/interface/LegacyTrackerTopology.h]: and maybe "call" -> "create" :-). Thanks!

Comment (alesaggio) [reply DataFormats/TrackerCommon/interface/LegacyTrackerTopology.h]: ops, fixing it :)

**Pull request merged**